File: cf_sdk/src/main/java/com/example/cf_sdk/logic/changebanksdk/source/AppPreferenceDatasource.java
package com.example.cf_sdk.logic.changebanksdk.source;

import io.reactivex.Completable;
import io.reactivex.Single;

/**
 *
 * Datasource for AppPreferences.
 */

public interface AppPreferenceDatasource {
    Single<Boolean> isWelcomeScreenCompletedOrSkipped();
    Completable setWelcomeScreenCompletedOrSkipped();
    Completable setHasLoggedInAlready();
    Completable clearAllAppPreference();
}

File: cf_sdk/src/main/java/com/example/cf_sdk/changebankapi/source/datasource/AppPreferenceDatasource.kt
package com.example.cf_sdk.changebankapi.source.datasource

import io.reactivex.Completable
import io.reactivex.Single

/**
 *
 * Datasource for AppPreferences.
 */
interface AppPreferenceDatasource {
    val isWelcomeScreenCompletedOrSkipped: Single<Boolean?>?
    fun setWelcomeScreenCompletedOrSkipped(): Completable?
    fun setHasLoggedInAlready(): Completable?
    fun clearAllAppPreference(): Completable?
}